src/core/thread/mle_router.cpp

```diff
@@ -2507,7 +2507,14 @@ void MleRouter::HandleChildUpdateRequest(const Message &         aMessage,
     }
 
 #if OPENTHREAD_CONFIG_MULTI_RADIO
-    child->ClearLastRxFragmentTag();
+    // We clear the fragment tag only if the "Child Update Request" is
+    // from a detached child trying to restore its link with its
+    // parent which is indicated by the presence of Challenge TLV in
+    // the message.
+    if (challenge.mLength != 0)
+    {
+        child->ClearLastRxFragmentTag();
```

Comment:
I wonder if the parent lose it's `fragment tag` during a quick reboot, how can the Child be aware and clear the last fragment tag.

Reply:
It is the same process again. The parent would send the "Child Update Request" to recover the child (child would be marked as in "restoring" state on parent node). 

In new pushed commit I added the same check in `mle.cpp` method (though we can also assume parent would often include "challenge" in Child Update Request" as I guess it mainly used for child.